modules/pam_unix/unix_chkpwd.c

```diff
@@ -103,6 +104,13 @@ int main(int argc, char *argv[])
 	 */
 	setup_signals();
 
+	/*
+	 * Use landlock to reduce the file scope of unix_chkpwd to /etc/shadow
+	 */
+
+	if (_restrict_to_file_read("/etc/shadow") > 0)
+		return PAM_SYSTEM_ERR;
```

Comment:
I guess such error should not be returned if Landlock is not supported by the running kernel, only if some legitimate Landlock calls failed unexpectedly.

Reply:
As the landlock support might be optional and thus explicitly asked by the user at configuration time this could lead to a hard dependency to it, which I think is a valid reason to return a `PAM_SYSTEM_ERR`, but I might be wrong.

Reply:
`PAM_SYSTEM_ERR` is legitimate but only when an unexpected Landlock error occurs, not when Landlock is not supported by the running kernel (best-effort security approach).

Reply:
Done, a LOG_NOTICE message is sent to the user and the program continues if Landlock is not supported by the running kernel.